samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatParticipantController.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using SemanticKernel.Service.CopilotChat.Hubs;
+using SemanticKernel.Service.CopilotChat.Models;
+using SemanticKernel.Service.CopilotChat.Storage;
+
+namespace SemanticKernel.Service.CopilotChat.Controllers;
+
+/// <summary>
+/// Controller for chat invitations.
+/// This controller is responsible for:
+/// 1. Creating invitation links.
+/// 2. Accepting/rejecting invitation links.
+/// 3. Managing participants in a chat session.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ChatParticipantController : ControllerBase
+{
+    private const string UserJoinedClientCall = "UserJoined";
+    private readonly ILogger<ChatParticipantController> _logger;
+    private readonly ChatParticipantRepository _chatParticipantRepository;
+    private readonly ChatSessionRepository _chatSessionRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatParticipantController"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="chatParticipantRepository">The chat participant repository.</param>
+    /// <param name="chatSessionRepository">The chat session repository.</param>
+    public ChatParticipantController(
+        ILogger<ChatParticipantController> logger,
+        ChatParticipantRepository chatParticipantRepository,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._logger = logger;
+        this._chatParticipantRepository = chatParticipantRepository;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Join a use to a chat session given a chat id and a user id.
+    /// </summary>
+    /// <param name="messageRelayHubContext">Message Hub that performs the real time relay service.</param>
+    /// <param name="chatParticipantParam">Contains the user id and chat id.</param>
+    [HttpPost]
+    [Route("chatParticipant/join")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> JoinChatAsync(
+        [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
+        [FromBody] ChatParticipant chatParticipantParam)
+    {
```

Comment:
How do we know a user was invited? It looks like if they know the chatID, we just accept them?

Reply:
Yes, that is the current design.